hollow/src/main/java/com/netflix/hollow/api/codegen/perfapi/HollowObjectTypePerfAPIClassGenerator.java

```diff
@@ -84,10 +84,14 @@ public void appendFieldMethod(StringBuilder builder, FieldType fieldType, String
         switch(fieldType) {
             case INT:
                 builder.append("    public int get" + HollowCodeGenerationUtils.upperFirstChar(fieldName) + "(long ref) {\n");
+                builder.append("        if(fieldIdx[" + fieldIdx + "] == -1)\n");
+                builder.append("            return Integer.MIN_VALUE;\n");
                 builder.append("        return typeAccess.readInt(ordinal(ref), fieldIdx[" + fieldIdx + "]);\n");
                 builder.append("    }\n\n");
 
                 builder.append("    public Integer get" + HollowCodeGenerationUtils.upperFirstChar(fieldName) + "Boxed(long ref) {\n");
+                builder.append("        if(fieldIdx[" + fieldIdx + "] == -1)\n");
+                builder.append("            return null;\n");
```

Comment:
Shouldn't this return `Integer.MIN_VALUE`? Just going by the classic generated API-
In `LongTypeAPI.java`,
```
public Long getValueBoxed(int ordinal) {
        long l;
        if(fieldIndex[0] == -1) {
            l = missingDataHandler().handleLong("Long", ordinal, "value");
        ...
```

Reply:
hmm interesting... in the actual generated code for something like `CislMidInsightsDelegateLookupImpl` the getter for primitive and boxed long looks like this: 

```java
    public long getMovieId(int ordinal) {
        ordinal = typeAPI.getMovieIdOrdinal(ordinal);
        return ordinal == -1 ? Long.MIN_VALUE : typeAPI.getAPI().getLongTypeAPI().getValue(ordinal);
    }

    public Long getMovieIdBoxed(int ordinal) {
        ordinal = typeAPI.getMovieIdOrdinal(ordinal);
        return ordinal == -1 ? null : typeAPI.getAPI().getLongTypeAPI().getValueBoxed(ordinal);
    }
``

Reply:
Hmm yeah looks like the defaults aren't consistently applied in the generated api

Reply:
Lets ship this improvement for now, I've tracked the fix for consistently applying defaults in [this issue](https://github.com/Netflix/hollow/issues/656)